include/nbl/builtin/hlsl/bxdf/base/cook_torrance_base.hlsl

```diff
@@ -131,33 +134,40 @@ struct SCookTorrance
     NBL_CONSTEXPR_STATIC_INLINE bool IsAnisotropic = ndf_type::IsAnisotropic;
     NBL_CONSTEXPR_STATIC_INLINE bool IsBSDF = ndf_type::NDFSurfaceType != ndf::MTT_REFLECT;
 
+    template<class Interaction, class MicrofacetCache>
+    static bool __checkValid(NBL_CONST_REF_ARG(fresnel_type) f, NBL_CONST_REF_ARG(sample_type) _sample, NBL_CONST_REF_ARG(Interaction) interaction, NBL_CONST_REF_ARG(MicrofacetCache) cache)
```

Comment:
why make `static` and pass the `f` if its your member?

Reply:
is it because the fresnel is already properly oriented? then give `f` a clear name like `reorientedFresnel`

and since its `static` you may as well make the `__checkValid` into a standalone struct functor with a partial spec for `IsBSDF`, that way you don't call `impl::check_TIR_helper` but you can call `cache.isValid(reorientedFresnel.getRefractionOrientedEta())` directly